__tests__/app.test.js

```diff
@@ -14,4 +25,47 @@ describe("GET /api", () => {
         expect(endpoints).toEqual(endpointsJson);
       });
   });
+  test("400: Responds with bad request if request made to invalid endpoint", () => {
+    return request(app)
+      .get("/api/invalid")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  describe("GET /api/topics", () => {
+    test("200: Responds with an array of topic objects, with the properties of slug and description", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body: { topics } }) => {
+          expect(topics.length).toBe(3);
+          topics.forEach((topic) => {
+            expect(topic).toMatchObject({
+              slug: expect.any(String),
+              description: expect.any(String),
+            });
+          });
+        });
+    });
+
+    test("404: Responds with 'No topics found' if no topics are found in the database", () => {
+      return db
+        .query("DELETE FROM comments")
+        .then(() => {
+          return db.query("DELETE FROM articles");
+        })
+        .then(() => {
+          return db.query("DELETE FROM topics");
```

Comment:
We shouldn't be doing any SQL database queries in the tests. This test also doesn't need to be here. The only thing that could go wrong at this point is if someone went to an incorrect endpoint (e.g. "/api/topic") but this wouldn't be exclusive to this endpoint and can go in it's own describe block for incorrect endpoints if you want to include it 👍